rskj-core/src/test/java/co/rsk/vm/PrecompiledContractAddressTests.java

```diff
@@ -0,0 +1,54 @@
+package co.rsk.vm;
+
+import co.rsk.config.TestSystemProperties;
+import co.rsk.core.RskAddress;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.apache.commons.lang3.builder.ToStringExclude;
+import org.ethereum.vm.DataWord;
+import org.ethereum.vm.PrecompiledContracts;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+/**
+ * Created by Sergio Demian Lerner on 12/10/2018.
+ */
+public class PrecompiledContractAddressTests {
+
+    public static final String ECRECOVER_ADDR = "0000000000000000000000000000000000000001";
+    public static final String SHA256_ADDR = "0000000000000000000000000000000000000002";
+    public static final String RIPEMPD160_ADDR = "0000000000000000000000000000000000000003";
+    public static final String IDENTITY_ADDR_STR = "0000000000000000000000000000000000000004";
+    public static final String BIG_INT_MODEXP_ADDR = "0000000000000000000000000000000000000005";
+    public static final String SAMPLE_ADDR_STR = "0000000000000000000000000000000001000005";
+    public static final String BRIDGE_ADDR_STR = "0000000000000000000000000000000001000006";
+    public static final String REMASC_ADDR_STR = "0000000000000000000000000000000001000008";
+
+    private final TestSystemProperties config = new TestSystemProperties();
+
+    @Test
+    public void testGetPrecompile() {
+        PrecompiledContracts pcList = new PrecompiledContracts(config);
+        checkAddr(pcList,ECRECOVER_ADDR, "ECRecover");
+        checkAddr(pcList,SHA256_ADDR, "Sha256");
+        checkAddr(pcList,RIPEMPD160_ADDR ,"Ripempd160");
+        checkAddr(pcList,IDENTITY_ADDR_STR ,"Identity");
+        checkAddr(pcList,BIG_INT_MODEXP_ADDR ,"BigIntegerModexp");
+        //checkAddr(pcList,SAMPLE_ADDR_STR ,"SamplePrecompiledContract");
+        checkAddr(pcList,BRIDGE_ADDR_STR ,"Bridge");
+        checkAddr(pcList,REMASC_ADDR_STR ,"RemascContract");
+    }
+
+    void checkAddr(PrecompiledContracts pcList,String addr,String className) {
+        RskAddress a;
+        a = new RskAddress(addr);
+        PrecompiledContracts.PrecompiledContract pc = pcList.getContractForAddress(null,new DataWord(a.getBytes()));
+        Assert.assertEquals(className,pc.getClass().getSimpleName());
+        // Do not get precompiles if the 12-byte prefix is non-zero.
+        DataWord dw = new DataWord(a.getBytes());
```

Comment:
Why not alter the byte array before construction?

Reply:
Yes, it could be better. However the VM code already alters the DataWord bytes externally (see the doBYTE).